### PR TITLE
feat: network state

### PR DESCRIPTION
**Pull request by jbutler**
**Issue #, if available:**

**Description of changes:**
Very simple network state infrastructure class which can be used to query probable network state.

**Why is this change necessary:**
Will become useful as we introduce more use cases which make network calls.

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java [hunk 1]
```diff
@@ -197,6 +200,11 @@ private void configChangeHandler(WhatHappened whatHappened, Node node) {
     private void registerEventListeners() {
         context.get(CACertificateChainChangedListener.class).listen();
         context.get(CAConfigurationChangedListener.class).listen();
+        context.get(CISShadowMonitor.class).listen();
+    }
+
+    private void initializeInfrastructure() {
```
Comment (jbutler) [inline src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java:206]: I hate this. Any better ideas for initializing infra pieces?

**Review by Nelsonochoam — APPROVED** [review]

File: src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import com.aws.greengrass.mqttclient.MqttClient;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnectionEvents;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.inject.Inject;
+
+public final class NetworkState {
+    public static final NetworkConnectivityState NETWORK_UP = NetworkConnectivityState.NETWORK_UP;
```
Comment (Nelsonochoam) [inline src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java:18]: Couldn't we reference this variables directly? They seem a bit redundant

File: src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    public static final NetworkConnectivityState NETWORK_DOWN = NetworkConnectivityState.NETWORK_DOWN;
```
Comment (Nelsonochoam) [inline src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java:19]: This variables seem a bit redundant maybe we can just reference the enum direclty

Comment (jbutler) [reply src/main/java/com/aws/greengrass/clientdevices/auth/infra/NetworkState.java]: I just did it as a shorthand. But that's fine. Removed and shorted `NetworkConnectivityState` to just `ConnectivityState`

**Review by Nelsonochoam — APPROVED** [review]

**Pull request merged**